File: web/src/api/system/role.ts
/**角色
 *
 */
import axiosReq from '@/utils/axios-req';

// 获取所有角色列表
export const getAllRole = async () => {
  return axiosReq({
    url: '/role/all',
    method: 'get',
    isParams: true,
    data: {},
  });
};

// 获取角色列表
export const getRoleList = async (params: any) => {
  return axiosReq({
    url: '/role/list',
    method: 'get',
    isParams: true,
    data: params,
  });
};

// 添加角色
export const addRole = async (params: any) => {
  return axiosReq({
    url: '/role/add',
    method: 'post',
    data: params,
  });
};

// 更新角色
export const updateRole = async (params: any) => {
  return axiosReq({
    url: '/role/update',
    method: 'put',
    data: params,
  });
};

// 删除角色
export const deleteRole = async (params: any) => {
  return axiosReq({
    url: '/role/delete',
    method: 'delete',
    data: params,
  });
};
// 批量删除角色
export const batchDeleteRole = async (params: any) => {
  return axiosReq({
    url: '/role/batchDelete',
    method: 'delete',
    data: params,
  });
};

// 更新角色状态
export const updateRoleStatus = async (params: any) => {
  return axiosReq({
    url: '/role/status',
    method: 'put',
    data: params,
  });
};

File: server/internal/router/api/v1/system/role.go
/*角色管理*/
package system

import (
	"gin-admin/internal/controller/system"

	"github.com/gin-gonic/gin"
)

// InitRoleRouter 初始化角色管理路由
func InitRoleRouter(group *gin.RouterGroup) {
	// 角色管理
	role := group.Group("/role")
	{
		// 获取所有角色列表
		role.GET("/all", system.NewRoleController().All)
		// 获取角色列表
		role.GET("/list", system.NewRoleController().List)
		// 添加角色
		role.POST("/add", system.NewRoleController().Add)
		// 更新角色
		role.PUT("/update", system.NewRoleController().Update)
		// 删除角色
		role.DELETE("/delete", system.NewRoleController().Delete)
		// 批量删除角色
		role.DELETE("/batchDelete", system.NewRoleController().BatchDelete)
		// 更新角色状态
		role.PUT("/status", system.NewRoleController().Status)
	}
}
